File: Project_Python3/ComputeHash.py
# coding: utf-8

import hashlib

class ComputeHash:

    # 関数配列の呼び出し
    def ComputeHash_Common(self, i, srcArray):
        if i == 0:
            return hashlib.md5(srcArray).hexdigest()
        elif i == 1:
            return hashlib.sha1(srcArray).hexdigest()
        elif i == 2:
            return hashlib.sha256(srcArray).hexdigest()
        elif i == 3:
            return hashlib.sha384(srcArray).hexdigest()
        elif i == 4:
            return hashlib.sha512(srcArray).hexdigest()
        else:
            return hashlib.sha256(srcArray).hexdigest()

File: Project_Golang/mysolution/searchcleartext/computehash/ComputeHash.go
package computehash

import (
	"crypto/md5"
	"crypto/sha1"
	"crypto/sha256"
	"crypto/sha512"
)

// ComputeHashCommon ...
// ハッシュ関数呼び出し用 for []byte
func ComputeHashCommon(AlgorithmIndex int, srcArray []byte) []byte {
	switch AlgorithmIndex {
	case 0:
		b := md5.Sum(srcArray)
		return b[:]
	case 1:
		b := sha1.Sum(srcArray)
		return b[:]
	case 2:
		b := sha256.Sum256(srcArray)
		return b[:]
	case 3:
		b := sha512.Sum384(srcArray)
		return b[:]
	case 4:
		b := sha512.Sum512(srcArray)
		return b[:]
	}

	return nil
}
